### DISCUSSION TITLE
NextJS x Keystatic - Migrate Keystatic API routes to /app/api folder

### DISCUSSION THREAD

**Discussion by emilienbidet** [Thinkmill/keystatic#321] [Ideas]
Hello !

I followed the step by step manual guide to add Keystatic to a NextJS project.

I was wondering why we can't use the new app directory to serve the Keystatic API.

<img width="870" alt="image" src="https://github.com/Thinkmill/keystatic/assets/44983373/4c9262bc-4972-425d-b438-7a8a1dc473a6">

I'm talking about this [docs](https://keystatic.com/docs/installation-next-js#keystatic-api-routes).

I would be happy to develop this feature.
Guide me on how could I help :)
Emilien

**Comment by simonswiss** [comment]
Hey Emilien!

Looks like you're now able to put everything inside the `app` directory with this PR:

https://github.com/Thinkmill/keystatic/pull/341

🎉

**Reply by emilienbidet** [reply]
OMG guys, I'm totally in ❤️ with KeyStatic !

Thank you very much 

How could I contribute?
